File: server/platform-check.ts
import './config.js';
import { pathToFileURL } from 'node:url';
import { publicConfig } from './config.js';
import { platformTools } from './connectors.js';
import { createRun, errorMessage } from './runtime.js';

export async function checkPlatforms() {
  return Promise.all((['erpnext', 'zammad'] as const).map(async source => {
    if (!publicConfig().connectors[source]) return { platform: source, status: 'not_configured', message: '尚未配置实例地址与 API 凭据' };
    const start = Date.now();
    try {
      const tools = platformTools(source);
      const run = createRun({ scenario: source === 'erpnext' ? 'finance' : 'support', source, mode: 'live', task: 'Read-only connectivity check' }, null);
      const context = { run, signal: AbortSignal.timeout(20000), evidence: new Set<string>() };
      const probes = source === 'erpnext' ? [{ name: 'erpnext_list_invoices', args: { status: 'all', page: 1, pageSize: 1 } }, { name: 'erpnext_list_payments', args: { page: 1, pageSize: 1 } }] : [{ name: 'zammad_list_tickets', args: { page: 1, pageSize: 1 } }, { name: 'zammad_list_states', args: {} }];
      for (const probe of probes) await tools.find(tool => tool.name === probe.name)!.execute(probe.args, context);
      return { platform: source, status: 'reachable', latencyMs: Date.now() - start, message: `已验证 ${probes.length} 个只读端点；数据初始化和完整权限仍需单独核查` };
    } catch (error) { return { platform: source, status: 'failed', latencyMs: Date.now() - start, message: errorMessage(error) }; }
  }));
}
if (process.argv[1] && import.meta.url === pathToFileURL(process.argv[1]).href) console.log(JSON.stringify(await checkPlatforms(), null, 2));

File: backend/platform_check.py
import asyncio
import time
from .config import public_config
from .connectors import platform_tools
from .runtime import create_run
from .tools import ToolContext


async def check_platforms():
    async def check(source):
        if not public_config()['connectors'][source]:
            return {'platform': source, 'status': 'not_configured', 'message': '尚未配置实例地址与 API 凭据'}
        started = time.monotonic()
        try:
            tools = {t.name: t for t in platform_tools(source)}
            context = ToolContext(create_run({'scenario': 'finance' if source == 'erpnext' else 'support', 'mode': 'live', 'source': source, 'task': 'Connection check'}))
            probes = [('erpnext_list_invoices', {'status': 'all', 'page': 1, 'pageSize': 1}), ('erpnext_list_payments', {'page': 1, 'pageSize': 1})] if source == 'erpnext' else [('zammad_list_tickets', {'page': 1, 'pageSize': 1}), ('zammad_list_states', {})]
            for name, args in probes:
                await tools[name].execute(args, context)
            return {'platform': source, 'status': 'reachable', 'latencyMs': round((time.monotonic() - started) * 1000), 'message': 'Python 连接器已验证两个只读端点'}
        except Exception as error:
            return {'platform': source, 'status': 'failed', 'message': str(error)[:500]}
    return await asyncio.gather(check('erpnext'), check('zammad'))


if __name__ == '__main__':
    import json
    print(json.dumps(asyncio.run(check_platforms()), ensure_ascii=False, indent=2))
